packages/mobile/src/screens/track-screen/DownloadSection.tsx

```diff
@@ -60,8 +61,15 @@ export const DownloadSection = ({ trackId }: { trackId: ID }) => {
   const { onOpen: openWaitForDownloadModal } = useWaitForDownloadModal()
   const [quality, setQuality] = useState(DownloadQuality.MP3)
   const [isExpanded, setIsExpanded] = useState(false)
+  const track = useSelector((state: CommonState) =>
+    getTrack(state, { id: trackId })
+  )
   const { stemTracks } = useCurrentStems({ trackId })
-  const shouldDisplayDownloadAll = stemTracks.length > 1
+  const { hasStreamAccess, hasDownloadAccess } = useGatedContentAccess(track)
+  // Hide the download all button if there aren't multiple downloads and if the user
+  // happens to not have stream access to the track
+  const shouldDisplayDownloadAll =
+    stemTracks.length > 1 && hasStreamAccess && hasDownloadAccess
```

Comment:
nit: not sure if `hasStreamAccess` is necessary here since `hasDownloadAccess` will never be true while `hasStreamAccess` is false, but doesn't hurt.

Reply:
yeah good callout -- I think it's helpful to be explicit, but I agree it's not possible as the backend currently allows it

Reply:
nit but i think for simplicity purposes, we can rely on just one boolean. that is one of the goals of this design so that UI doesnt have to do a whole lot of thinking

Reply:
ok sounds good, will make that change